Patch-Source:
https://github.com/Vimjas/vint/commit/997677ae688fbaf47da426500cc56aae7305d243
https://github.com/Vimjas/vint/commit/ad2fef93c90b9f5acf7df42619c6111630afc072
Use setuptools_scm for version handling and drop deprecated pkg_resources.
Backported for 0.3.21.
---
diff -rupN a/setup.py b/setup.py
--- a/setup.py	2019-05-13 23:10:24.000000000 +0000
+++ b/setup.py	2025-11-11 15:40:20.220000000 +0000
@@ -1,6 +1,5 @@
 #!/usr/bin/env python
 import sys
-import os.path
 from setuptools import setup, find_packages
 
 
@@ -31,23 +30,19 @@ def test_requires():
     return requires
 
 
-def get_version():
-    vint_root = os.path.dirname(__file__)
-    version_file = open(os.path.join(vint_root, 'VERSION.txt'))
-    return version_file.read().strip()
-
-
-VERSION = get_version()
-
-
 setup(
     name='vim-vint',
-    version=VERSION,
     description='Lint tool for Vim script Language',
     author='Kuniwak',
     author_email='orga.chem.job+vint@gmail.com',
     url='https://github.com/Kuniwak/vint',
-    download_url='https://github.com/Kuniwak/vint/archive/v{version}.tar.gz'.format(version=VERSION),
+    download_url='https://github.com/Kuniwak/vint/releases',
+    use_scm_version={
+        'write_to': 'vint/__version__.py',
+    },
+    setup_requires=[
+        'setuptools_scm',
+    ],
     install_requires=install_requires(),
     tests_require=test_requires(),
     packages=find_packages(exclude=['dev_tool', 'test*']),
diff -rupN a/vint/linting/cli.py b/vint/linting/cli.py
--- a/vint/linting/cli.py	2019-05-13 23:10:24.000000000 +0000
+++ b/vint/linting/cli.py	2025-11-11 15:36:46.880000000 +0000
@@ -1,7 +1,6 @@
 import sys
 from argparse import ArgumentParser
 from pathlib import PosixPath
-import pkg_resources
 import logging
 
 from vint.linting.linter import Linter
@@ -150,13 +149,7 @@ class CLI(object):
 
 
     def _get_version(self):
-        # In unit tests, pkg_resources cannot find vim-vint.
-        # So, I decided to return dummy version
-        try:
-            version = pkg_resources.require('vim-vint')[0].version
-        except pkg_resources.DistributionNotFound:
-            version = 'test_mode'
-
+        from ..__version__ import version
         return version
 
 
